### PR TITLE
Make DnDv2 themable

### PR DESCRIPTION
**Pull request by itsjeyd**
This PR adds the necessary infrastructure for applying custom themes to DnDv2. It also makes legacy (Apros) styling of the block available as an opt-in theme.

**Screenshot**

Appearance of a DnDv2 exercise with legacy theme enabled:

apros-theme

**Testing**

1. Verify that DnDv2 uses default colors when legacy theme is not enabled.
2. Follow instructions in updated README to enable legacy theme, then verify that DnDv2 uses styles shown in screenshot above.

### CODE REVIEW

File: drag_and_drop_v2/templates/html/js_templates.html [hunk 1]
```diff
@@ -123,7 +123,7 @@
         </div>
         <div class="row">
           <label for="item-{{id}}-numerical-margin">
-            {{i18n "Margin ± (when a numerical value is required, values entered by students must not differ from the expected value by more than this margin; default is zero)"}}
+            {{i18n "Margin +/- (when a numerical value is required, values entered by students must not differ from the expected value by more than this margin; default is zero)"}}
```
Comment (bradenmacdonald) [inline drag_and_drop_v2/templates/html/js_templates.html]: Why this change? Personally I prefer the correct symbol, but don't have a strong opinion.

File: setup.py [hunk 2]
```diff
@@ -31,7 +31,6 @@ def package_data(pkg, root_list):
         'xblock-utils',
         'ddt'
     ],
-    dependency_links = ['http://github.com/edx/xblock-utils/tarball/master#egg=xblock-utils'],
```
Comment (bradenmacdonald) [inline setup.py]: Something has broken the Travis build. I think it might be this change?

File: drag_and_drop_v2/public/themes/apros.css [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+.themed-xblock.xblock--drag-and-drop .popup .popup-content {
+    color: #ffffff;
+    font-size: 14px;
+}
+
+.themed-xblock.xblock--drag-and-drop .popup .close {
+    cursor: pointer;
+    color: #ffffff;
+    font-family: "fontawesome";
+    font-size: 18pt;
```
Comment (bradenmacdonald) [inline drag_and_drop_v2/public/themes/apros.css]: These rules (and probably others in this file) are identical to the unthemed version - can you remove them from this file? (It's good that these things can be overridden by the theme, but we don't need to actually include new values for rules that aren't different here in the apros theme.)

File: drag_and_drop_v2/public/themes/apros.css [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+/*** FEEDBACK ***/
+
+.themed-xblock.xblock--drag-and-drop .feedback {
+    border-top: solid 1px #bdbdbd;
+}
+
+.themed-xblock.xblock--drag-and-drop .popup {
+    background: none repeat scroll 0 0 #66a5b5;
```
Comment (bradenmacdonald) [inline drag_and_drop_v2/public/themes/apros.css]: I think you can just use `background-color: #66a5b5;` here and it has the same effect.

File: drag_and_drop_v2/templates/html/js_templates.html (hunk 1 above)
Comment (itsjeyd) [inline drag_and_drop_v2/templates/html/js_templates.html]: @bradenmacdonald 

After replacing `load_resource` with `ResourceLoader.load_unicode` in the [line](https://github.com/open-craft/xblock-drag-and-drop-v2/blob/add-theming/drag_and_drop_v2/drag_and_drop_v2.py#L180) that pulls in `js_templates.html` for the EDIT dialog, I got the following error:

    Error: 'ascii' codec can't decode byte 0xc2 in position 5557: ordinal not in range(128)

The difference between `load_resource` and `load_unicode` from xblock-utils is that the former wasn't calling `unicode` on the content pulled from a resource before returning it.

Changing the symbol seemed like a better option than keeping `load_resource` just for the sake of avoiding this error, but let me know -- maybe there's another way to approach this?

File: drag_and_drop_v2/public/themes/apros.css (hunk 3 above)
Comment (itsjeyd) [inline drag_and_drop_v2/public/themes/apros.css]: @bradenmacdonald The reason I'm making these styles explicit here is to make sure they stay the same even if the default styles change. Based on your comment I'm not completely sure if that's a valid concern? If you think it's unlikely that the defaults will change in the future, let me know and I'll remove the rules/declarations that are identical to the unthemed version.

Comment (bradenmacdonald) [inline drag_and_drop_v2/public/themes/apros.css]: @itsjeyd Ok, fair enough.

File: drag_and_drop_v2/templates/html/js_templates.html (hunk 1 above)
Comment (bradenmacdonald) [inline drag_and_drop_v2/templates/html/js_templates.html]: @itsjeyd Hmm, you'd think a method called `load_unicode` would support loading unicode text :-p

Your workaround is fine for now, so we can get this merged on time. But the proper fix is to change xblock-utils to use `return unicode(resource_content, 'utf-8')` instead of just `return unicode(resource_content)`. Would you be able to open a second PR to fix that in xblock-utils?

Comment (itsjeyd) [inline drag_and_drop_v2/templates/html/js_templates.html]: Already mentioned this on OC-1164 but for reference, https://github.com/edx/xblock-utils/pull/36 implements the fix for this issue.

**Comment by bradenmacdonald** [comment]
:+1: Great work @itsjeyd !

Let's merge this now. I've made a note on the big PR, #34, that we can restore the "±" character if the xblock-utils fix makes it into Dogwood.
